dask_planner/src/parser.rs

```diff
@@ -164,6 +182,44 @@ impl<'a> DaskParser<'a> {
                         // use custom parsing
                         self.parse_drop()
                     }
+                    Keyword::SELECT => {
+                        // Check for PREDICT token in statement
+                        let mut cnt = 1;
+                        loop {
+                            match self.parser.next_token() {
+                                Token::Word(w) => {
+                                    match w.value.to_lowercase().as_str() {
+                                        "predict" => {
+                                            return self.parse_predict_model();
+                                        }
+                                        _ => {
+                                            // Keep looking for PREDICT
+                                            cnt += 1;
+                                            continue;
+                                        }
+                                    }
+                                }
+                                Token::EOF => {
+                                    break;
+                                }
+                                _ => {
+                                    // Keep looking for PREDICT
+                                    cnt += 1;
+                                    continue;
+                                }
+                            }
+                        }
+
+                        // Reset the parser back to where we started
```

Comment:
Wanted to use `self.parser.index = 0` but `index` is private so had to resort to this abomination